src/server/helpers/metabrainzNotifications.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable camelcase */
+import {cacheJSON, getCachedJSON} from '../../common/helpers/cache';
+import config from '../../common/helpers/config';
+
+import log from 'log';
+import request from 'superagent';
+
+
+const notificationScopes = ['notification'];
+const TOKEN_CACHE_KEY = 'notification_access_token';
+
+/**
+ * Fetches a valid OAuth2 access token with the notification scope for the MetaBrainz notification API.
+ * If a cached token exists in Redis, it’s used, else a new one is requested.
+ * @returns {Promise<string|null>} OAuth2 access token or null if failed
+ */
+async function getMetaBrainzNotificationToken(): Promise<string | null> {
+	const cachedToken = await getCachedJSON<string>(TOKEN_CACHE_KEY);
+	if (cachedToken) {
+		return cachedToken;
+	}
+
+	const {clientID, clientSecret, tokenURL, notificationURL} = config.oauth;
+
+	try {
+		const response = await request
+			.post(tokenURL)
+			.type('form')
+			.send({
+				client_id: clientID,
+				client_secret: clientSecret,
+				grant_type: 'client_credentials',
+				scope: notificationScopes
+			});
+
+		const {access_token: accessToken, expires_in: expiresIn} = response.body;
+		await cacheJSON(TOKEN_CACHE_KEY, accessToken, {expireTime: expiresIn});
+		return accessToken;
+	}
+	catch (error: any) {
+		log.error(error);
+		return null;
+	}
+}
+
+/**
+ * Sends multiple notifications in one request.
+ * @param {NotificationT[]} notifications - Array of notifications to send
+ * @returns {Promise<void>} Resolves when the notifications has been sent.
+ */
+export async function sendMultipleNotifications(notifications: NotificationT[]): Promise<void> {
+	const token = await getMetaBrainzNotificationToken();
+	const url = `${notificationURL}/send`;
```

Comment:
Unless I'm reading it wrong, you don't currently have access to the notificationURL in the scope of this function, and you're missing `const const {notificationURL} = config.oauth;`